File: Intersection_of_Two_Arrays_II/Intersection_of_Two_Arrays_II.cpp
class Solution
{
public:
    vector<int> intersect(vector<int> &nums1, vector<int> &nums2)
    {
        unordered_map<int, int> h1;
        unordered_map<int, int> h2;

        for (int x : nums1)
        {
            h1[x]++;
        }

        for (int x : nums2)
        {
            h2[x]++;
        }
        vector<int> result;
        for (auto i : h1)
        {
            if (h2.find(i.first) != h2.end())
            {

                int frequent;
                frequent = min(h2.at(i.first), i.second);
                for (int j = 0; j < frequent; j++)
                {
                    result.push_back(i.first);
                }
            }
        }
        return result;
    }
};

File: Intersection_of_Two_Arrays_II/Intersection_of_Two_Arrays_II.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        def hash(nums):
            hashmap = {}
            for n in nums:
                if n not in hashmap:
                    hashmap[n] = 0
                hashmap[n] = hashmap.get(n, 0) + 1
            return hashmap

        h1 = hash(nums1)
        h2 = hash(nums2)
        result = []
        for i in h1:
            if i in h2:
                frequent = min(h1[i], h2[i])
                for j in range(frequent):
                    result.append(i)
        return result
